lib/plugins/aws/package/compile/events/apiGateway/lib/restApi.js

```diff
@@ -45,6 +54,7 @@ module.exports = {
           BinaryMediaTypes,
           EndpointConfiguration: {
             Types: [endpointType],
+            VpcEndpointIds: [endpointVpcIds],
```

Comment:
Let's not assign this property unless it's set in a configuration (in some previous cases, where we allowed defaults to be forced into configuration it introduced nasty regressions).

This also currently crashes tests

Reply:
So what would be a best approach to do that?

Something like:

`
_.merge(this.serverless.service.provider.compiledCloudFormationTemplate.Resources, {
      [this.apiGatewayRestApiLogicalId]: {
        Type: 'AWS::ApiGateway::RestApi',
        Properties: {
          Name: this.provider.naming.getApiGatewayName(),
          BinaryMediaTypes,
          EndpointConfiguration: endpointVpcIds
            ? {
                Types: [endpointType],
                VpcEndpointIds: [endpointVpcIds],
              }
            : {
                Types: [endpointType],
              },
        },
      },
    });
`

Reply:
Most solid will probably be:

```
VpcEndpointIds: endpointVpcIds != null ? [endpointVpcIds] : undefined
```

Assigning `undefined` will make this properly automatically discarded for JSON.

One question though: Var name ` endpointVpcIds` suggests plural. Does it mean we already may deal with list of VPC endpoint ids? If so, is configuring it as `[endpointVpcIds]` safe?

Reply:
Yes. It can be plural but must be a string according AWS Documentation:

https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-apigateway-restapi-endpointconfiguration.html

Reply:
I was refering to `endpointVpcIds` var. Plural name suggests it's an array, not a string.

If it's an array, then with as proposed configuration what will end in `VpcEndpointIds` is `[[endpointVpcId1, endpointVpcId2]]`, instead of `[endpointVpcId1, endpointVpcId2]`

Reply:
It's a string but can be multiples vpc`s separate by comma inside of a string. But I got you in the code perspective is just a string. I'll change that as well. Thanks @medikoo

Reply:
Ok, so AWS supports format as: `['vpcEndpointId1, vpcEndpointId1']` ? Note It's an array with one item string, where string contains mutlitple VPC endpoints separated with coma.

Reply:
Yes! 

Like this:

`
{
  "Types" : [ String, ... ],
  "VpcEndpointIds" : [ String, ... ]
}
`

Reply:
Ok, so I guess currently we got it wrong. Currently configuration will make it as:

```
"VpcEndpointIds" : [ "vpcEndpointId1, vpcEndpointId2" ]
```

While it needs to be:

```
"VpcEndpointIds" : [ "vpcEndpointId1", "vpcEndpointId2" ]
```

Or do I miss something?

Reply:
Good question. I think the correct away would be the second one (following the AWS Documentation). If we're right the "Types" property is wrong too because the "Types" property has the same type of "VpcEndpointIds" -> "List of String".

Look:

> Properties
Types
A list of endpoint types of an API or its custom domain name. Valid values include:
EDGE: For an edge-optimized API and its custom domain name.
REGIONAL: For a regional API and its custom domain name.
PRIVATE: For a private API.
Required: No
Type: List of String
Update requires: No interruption
VpcEndpointIds
A list of VPC endpoint IDs of an API (AWS::ApiGateway::RestApi) against which to create Route53 ALIASes. It is only supported for PRIVATE endpoint type.
Required: No
Type: List of String
Update requires: No interruption

Font: https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-apigateway-restapi-endpointconfiguration.html#cfn-apigateway-restapi-endpointconfiguration-vpcendpointids

Reply:
> I think the correct away would be the second one

Yes, this spec clearly indicate they're after array of strings (and not one string that contains comma separated ids)

Reply:
So the current Types property it's wrong in this version.